wallet-new/src/Cardano/Wallet/Server/Plugins.hs

```diff
@@ -69,11 +79,32 @@ apiServer protocolMagic (NewWalletBackendParams WalletBackendParams{..}) (passiv
             (BS8.unpack ip)
             port
             (if isDebugMode walletRunMode then Nothing else walletTLSParams)
-            Nothing
+            (Just $ setOnExceptionResponse exceptionHandler defaultSettings)
             (Just $ portCallback ctx)
   where
     (ip, port) = walletAddress
 
+    exceptionHandler :: SomeException -> Response
+    exceptionHandler se = case translateKernelErrors se of
+            Just we -> handleLayerError we
+            Nothing -> handleGenericError se
+
+    -- Handle domain-specific errors coming from the Wallet Layer
+    handleLayerError :: V1.WalletError -> Response
+    handleLayerError we =
+            responseLBS (V1.toHttpErrorStatus we) [applicationJson] . encode $ we
+
+    -- Handle general exceptions
+    handleGenericError :: SomeException -> Response
+    handleGenericError se =
+        responseLBS badRequest400 [applicationJson] $ encode defWalletError
+        where
+            -- NOTE: we expose the entire exception, which we here assume reveals
+            --       no sensitive information. If that assumption is wrong, we can
+            --       use Typeable to inspect constructor names, which we can reveal here
+            --       wihle omitting the constructor arguments
+            defWalletError = V1.UnknownError $ T.pack (show se)
```

Comment:
This is one of those things that I keep going back and forth to and I think that at the cost of losing information, let's go with the conservative approach: let's do a `T.pack $ show $ typeOf se` to show just the type, and now the precise exception. Unfortunately I wasn't thinking properly the other day and obviously with this approach we cannot show the constructor names, for that we would need a `Generic` instance I believe. Typeable just gives us the type name, but I think that's better than a security breach.